django/db/migrations/state.py

```diff
@@ -548,7 +549,7 @@ def render(self, apps):
         body.update(self.construct_managers())
 
         # Then, make a Model object (apps.register_model is called in __new__)
-        return type(
+        return metaclassmaker()(
```

Comment:
It could be even done by `try except` and use `metaclassmaker` only when there would be some problems. It would slow down "this few projects with metaclasses" ;) but there won't be much slowdown for the other folks.